### DISCUSSION TITLE
Updating action/view post model creation throws `trap returned falsish for property`

### DISCUSSION THREAD

**Discussion by VitaliiK91** [mobxjs/mobx-state-tree#1709] [Q&A]
I need to update/create views and actions on the model after the model has been created. Redefine view/action after creation works when I am in dev mode (webpack), however it fails in production mode when I set new value of the view/action with the error:
```
TypeError: 'set' on proxy: trap returned falsish for property 'custom'
    at Configuration.store.ts:107
    at Generator.next (<anonymous>)
    at mobx-state-tree.module.js:4397
    at Ie (mobx.esm.js:1086)
    at <unnamed action> (mobx.esm.js:1070)
    at mobx-state-tree.module.js:3235
    at N (mobx-state-tree.module.js:3078)
    at u (mobx-state-tree.module.js:4383)
    at c (mobx-state-tree.module.js:4397)
```

ConfigurationStore is where I set new values for the model view/action

Is it expected? And if so why and what workarounds are there?
Thanks!

**Comment by EmilTholin** [comment]
Hi @VitaliiK91!

Very interesting. Do you think you could share a code example where you run into this? Maybe there is another way to do it, or maybe there is a bug in MST.

**Reply by VitaliiK91** [reply]
I think I figured a way around the issue. If I don't set view/action but rather create a closure variable and create an action to set it and then return it works. Here is an example of workaround:
```
MyModel
.views((self) => {
        let virtual: Function = () => { return undefined };
        return {
            setVirtual(f: Function) {
                virtual = f;
            },
            virtual: function(path: string) {
                return virtual(path);
            },
         }
```

**Reply by VitaliiK91** [reply]
I can post an example of how to reproduce issue, but a bit later. Thanks!
